odoo/addons/base/tests/test_ir_actions.py

```diff
@@ -383,3 +383,63 @@ def test_unlink_with_dependant(self):
         # uninstall mode: unlink dependant fields
         field.with_context(_force_unlink=True).unlink()
         self.assertFalse(dependant.exists())
+
+    def test_related_field(self):
+        """ create a related field, check filled values"""
+        SOURCE_MODEL = 'res.users'
+        SOURCE_FIELD = 'login'
+
+        TARGET_MODEL = 'res.partner'
+        target_model = self.env['ir.model'].search([('model', '=', TARGET_MODEL)])
+
+        target_records = self.env[TARGET_MODEL]
+        source_records = self.env[SOURCE_MODEL]
+        for i in range(10):
+            # create a user
+            source = self.env[SOURCE_MODEL].create({
+                'name': 'salesperson %s' % i,
+                SOURCE_FIELD: 'value_%s' % i,
+            })
+            # create partner with user_id (Salesperson) field
+            target = self.env[TARGET_MODEL].create({
+                'name': '%s' % i,
+                'user_id': source.id,
+            })
+            source_records |= source
+            target_records |= target
+
+        if odoo.release.version_info[0] >= 13:
+            source_records.flush()
+            target_records.flush()
+
+        if self.env[TARGET_MODEL].with_context(active_test=False).search_count([]) > odoo.models.PREFETCH_MAX:
+            self.skipTest("Too much records in model %s" % TARGET_MODEL)
+
+        sql_log = self.env.cr.sql_log
+        self.env.cr.sql_log = True
+        # Add custom field equivalent to the following definition:
+        #
+        # class Partner(models.Model)
+        #     _inhert = 'res.partner'
+        #     x_oh_boy = fields.Char(related="user_id.login", store=True)
+        self.env['ir.model.fields'].create({
+            'model_id': target_model.id,
+            'name': 'x_oh_boy',
+            'field_description': 'x_oh_boy',
+            'ttype': 'char',
+            'related': 'user_id.%s' % SOURCE_FIELD,
+            'store': True,
+        })
+        for target in target_records:
+            self.assertEqual(target.x_oh_boy, 'value_%s' % target.name)
+
+        self.env.cr.sql_log = sql_log
+        asserts = (
+            ('sql_update_log', 1),
+            ('sql_from_log', 3),
+            ('sql_into_log', 1),
+        )
+        TABLE = 'res_partner'
+        for t, expected_value in asserts:
+            real_value = self.env.cr.sql_update_log.get(TABLE, [0])[0]
+            self.assertLessEqual(real_value, expected_value, "%s for %s is not good" % (t, TABLE))
```

Comment:
Note that the test should make a diff before/after with those values.
The test fails if you launch it with debugging turned on.